test/mitmproxy/tools/console/test_commander.py

```diff
@@ -1,9 +1,35 @@
+import os
 import pytest
+import shutil
+import uuid
 
+from mitmproxy import options
+from mitmproxy.addons import command_history
 from mitmproxy.test import taddons
 from mitmproxy.tools.console.commander import commander
 
 
+@pytest.fixture(autouse=True)
+def tctx():
+    # This runs before each test
+    dir_id = str(uuid.uuid4())
+    confdir = os.path.expanduser(f"~/.mitmproxy-test-suite-{dir_id}")
```

Comment:
`~/.mitmproxy-test-suite` could be a bit annoying. http://doc.pytest.org/en/latest/tmpdir.html should be the way to go.

Reply:
Does it matter? The directory is deleted at the end of the test, so I thought that name would be fine.

Reply:
Probably not too much, but there are two arguments for `tmpdir`:

- Consistency: We already use `tmpdir` at other places. `tmpdir` gracefully handles test failures etc.
- Temporary files belong into the system temporary directory. 🙃

Reply:
Fixed!